adapters/repos/db/vector/hnsw/index.go

```diff
@@ -170,17 +171,24 @@ type hnsw struct {
 	// to define the rescoring concurrency.
 	rescoreConcurrency int
 
-	compressActionLock *sync.RWMutex
-	className          string
-	shardName          string
-	VectorForIDThunk   common.VectorForID[float32]
-	shardedNodeLocks   *common.ShardedRWLocks
-	store              *lsmkv.Store
+	compressActionLock       *sync.RWMutex
+	className                string
+	shardName                string
+	VectorForIDThunk         common.VectorForID[float32]
+	MultipleVectorForIDThunk common.MultipleVectorForID[float32]
+	shardedNodeLocks         *common.ShardedRWLocks
+	store                    *lsmkv.Store
 
 	allocChecker            memwatch.AllocChecker
 	tombstoneCleanupRunning atomic.Bool
 
 	visitedListPoolMaxSize int
+
+	multivector    atomic.Bool
+	vectorDocIDMap map[uint64]uint64
+	docIDVectorMap map[uint64][]uint64
```

Comment:
Ideally, we should avoid naming variables with the type they are in.

Example: 
- `vectorDocIDMap` -> `vectorDocIDs`
- `relativeIDMap` -> `relativeIDs`